From 4f0de62b201c413736c07605d123edc4366a939e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 11 Jan 2024 23:49:20 +0100
Subject: [PATCH 0430/1695] drivers: regulator: DT properties
 active-discharge/over-current-protection

Define regulator property flags for the regulator DT binding
properties regulator-over-current-protection and
regulator-active-discharge.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 05484432284d2133daea09cb75ce0782a030ecaf drivers: regulator: DT properties active-discharge/over-current-protection]
Change-Id: Iac4a008634235eb8aeeb02194850ba4751005d55
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/drivers/regulator/regulator_dt.c |  8 ++++++++
 core/include/drivers/regulator.h      | 10 +++++++++-
 2 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/core/drivers/regulator/regulator_dt.c b/core/drivers/regulator/regulator_dt.c
index 426541908..8b2c1a8da 100644
--- a/core/drivers/regulator/regulator_dt.c
+++ b/core/drivers/regulator/regulator_dt.c
@@ -40,6 +40,14 @@ static struct regulator_property flag_prop[] = {
 		.name = "regulator-boot-on",
 		.flag = REGULATOR_BOOT_ON,
 	},
+	{
+		.name = "regulator-over-current-protection",
+		.flag = REGULATOR_OVER_CURRENT,
+	},
+	{
+		.name = "regulator-active-discharge",
+		.flag = REGULATOR_ACTIVE_DISCHARGE,
+	},
 };
 
 /*
diff --git a/core/include/drivers/regulator.h b/core/include/drivers/regulator.h
index c9266bf32..81061b8da 100644
--- a/core/include/drivers/regulator.h
+++ b/core/include/drivers/regulator.h
@@ -26,9 +26,17 @@
  * DT property: regulator-boot-on
  */
 #define REGULATOR_BOOT_ON	BIT(2)
+/*
+ * Enables over current protection.
+ * DT property: regulator-over-current-protection
+ */
+#define REGULATOR_OVER_CURRENT	BIT(3)
+/* Enable active discharge. DT property: regulator-active-discharge */
+#define REGULATOR_ACTIVE_DISCHARGE	BIT(4)
 
 #define REGULATOR_FLAGS_MASK	(REGULATOR_ALWAYS_ON | REGULATOR_PULL_DOWN | \
-				 REGULATOR_BOOT_ON)
+				 REGULATOR_BOOT_ON | REGULATOR_OVER_CURRENT | \
+				 REGULATOR_ACTIVE_DISCHARGE)
 
 struct regulator_ops;
 
-- 
2.47.2

